### ISSUE TITLE
Test functions on custom test source sets are treated as non test functions

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [detekt/detekt#5896]
## Expected Behavior
Any test function should allow to use backticks and spaces even if it is not on an "official" test source set

## Observed Behavior
I have created custom source sets for different type tests, `testIntegration` and `testFunctional` and the functions reported on those source sets are marked by detekt, for example:

```kotlin
@Test
fun `android configuration cache clean v1_0_0`() { // marked by detekt
    gradleTestKitTest("gradle-features/android configuration cache clean v1_0_0") {
        beforeTest()
        testConfigurationCache(expectTaskOutcome = TaskOutcome.SUCCESS)
    }
}
```

## Steps to Reproduce
The repo with the function mentioned above can be found [here](https://github.com/JavierSegoviaCordoba/semver-gradle-plugin/blob/a46c34ccd5a462c3cf5750bbd2fe6a3180e32713/semver-project-gradle-plugin/testFunctional/kotlin/com/javiersc/semver/project/gradle/plugin/GradleFeaturesTest.kt)

---

I know it is not a good solution, but Detekt should check if a source set contains the word `test` or `Test` in order to apply some rules or others as, I think, it is impossible to know if a custom source set or configuration is for tests or not.

**Comment by BraisGabin** [comment]
I asume that you are using the default configuration. There that rule have this configuration:

```yaml
    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/androidUnitTest/**', '**/androidInstrumentedTest/**', '**/jsTest/**', '**/iosTest/**']
```

As you can see `functionalTest` is not listed there so it doesn't apply. You can just add it and everything should work as expected.

**Comment by schalkms** [comment]
Since there is nothing actionable here, I close this issue. If not, please leave a comment.

**Comment by JavierSegoviaCordoba** [comment]
I missed this notification.

I think the rule should find for *Test instead of hardcoding some of the multiplatform targets.

@schalkms @BraisGabin should I open a different issue or this is valid for this feature request?

**Comment by schalkms** [comment]
@JavierSegoviaCordoba If #5412 does not already fit for your use case, please add your thoughts to this issue.

**Comment by schalkms** [comment]
@andersfischernielsen to my knowledge, there is no merging but rather a overriding mechanism with `buildUponDefaultConfig = true`.

**Comment by andersfischernielsen** [comment]
That's a shame - I was hoping we could extend the existing excludes rather than replace them. But thanks nonetheless!

**Comment by schalkms** [comment]
@andersfischernielsen related discussion, which essentially proposes a solution to your scenario.
https://github.com/detekt/detekt/issues/5977#issuecomment-1522218952
